src/tooltip-manager.coffee

```diff
@@ -129,19 +130,37 @@ class TooltipManager
 
     tooltip = new Tooltip(target, options, @viewRegistry)
 
+    if !@tooltips.has(target)
+      @tooltips.set(target, [])
+    @tooltips.get(target).push(tooltip)
+
     hideTooltip = ->
       tooltip.leave(currentTarget: target)
       tooltip.hide()
 
     window.addEventListener('resize', hideTooltip)
 
-    disposable = new Disposable ->
+    disposable = new Disposable =>
       window.removeEventListener('resize', hideTooltip)
       hideTooltip()
       tooltip.destroy()
 
+      if @tooltips.has(target)
+        tooltipsForTarget = @tooltips.get(target)
+        index = tooltipsForTarget.indexOf(tooltip)
+        if index != -1
+          tooltipsForTarget.splice(index, 1)
+        if tooltipsForTarget.length == 0
+          @tooltips.delete(target)
+
     disposable
 
+  findTooltips: (target) ->
+    if @tooltips.has(target)
+      @tooltips.get(target)
```

Comment:
I noticed that other implementations of the `find` pattern use `slice` to create a shallow copy of the array and return the copy. I assume they're doing that to prevent people from corrupting the source array. Can we use `@tooltips.get(target).slice()` here?